src/CollectdWinService/WriteHttpPlugin.cs

```diff
@@ -140,25 +168,55 @@ public void HttpPost(string metricJsonStr)
 
                 using (Stream reqStream = request.GetRequestStream())
                 {
+                    if (Logger.IsTraceEnabled)
+                    {
+                        Logger.Trace("Adding request body : {0}", metricJsonStr);
+                    }
+                    
                     reqStream.Write(data, 0, data.Length);
                 }
 
                 response = (HttpWebResponse) request.GetResponse();
-                var statusCode = (int) response.StatusCode;
-                if (statusCode < 200 || statusCode >= 300)
+
+                // Skip overhead of the trace body read
+                if (Logger.IsTraceEnabled) 
                 {
-                    Logger.Error("Got a bad response code : ", statusCode);
+                    Stream respStream = response.GetResponseStream();
+                    string responseString = new StreamReader(respStream).ReadToEnd();
+                    Logger.Trace("Got response: {0}" + responseString);
                 }
+            }
+            catch (WebException ex)
+            {
+                if (ex.Status == WebExceptionStatus.ProtocolError)
+                {
+                    HttpWebResponse exceptionResponse = (HttpWebResponse)ex.Response;
 
-                if (!Logger.IsTraceEnabled) return;
+                    Logger.Error("Got web exception in http post : {0} - {1}",
+                            (int)exceptionResponse.StatusCode, exceptionResponse.StatusCode);
 
-                Stream respStream = response.GetResponseStream();
-                string responseString = new StreamReader(respStream).ReadToEnd();
-                Logger.Trace("Got response: " + responseString);
+                    if (!Logger.IsTraceEnabled)
+                    {
+                        // Skip overhead of trace body read 
+                        using (var stream = exceptionResponse.GetResponseStream())
+                        using (var reader = new StreamReader(stream))
+                        {
+                            string errorBody = reader.ReadToEnd();
+                            if (errorBody != null)
+                            {
+                                Logger.Error(errorBody);

```

Comment:
Probably, argh. I'm in Sal Paulo with no laptop. Can I check that in the morning?

Thx!

Patrick
(via mobile)

> On Oct 28, 2015, at 5:36 PM, Yoga Ramalingam <notifications@github.com> wrote:
> 
> In src/CollectdWinService/WriteHttpPlugin.cs:
> 
> >  
> > -                Stream respStream = response.GetResponseStream();
> > -                string responseString = new StreamReader(respStream).ReadToEnd();
> > -                Logger.Trace("Got response: " + responseString);
> > +                    if (!Logger.IsTraceEnabled)
> > +                    {
> > +                        // Skip overhead of trace body read 
> > +                        using (var stream = exceptionResponse.GetResponseStream())
> > +                        using (var reader = new StreamReader(stream))
> > +                        {
> > +                            string errorBody = reader.ReadToEnd();
> > +                            if (errorBody != null)
> > +                            {
> > +                                Logger.Error(errorBody);
> should this be Logger.Trace ?
> 
> —
> Reply to this email directly or view it on GitHub.
>